repl/repl.go

```diff
@@ -232,20 +247,24 @@ func Interactive(options Options) int {
 				"Type 'history' to see history, '!n' to repeat history n, 'info' for language builtins, use <tab> for completion.")
 			continue
 		}
-		// errors are already logged and this is the only case that can get contNeeded (EOL instead of EOF mode)
-		contNeeded, _, formatted := EvalOne(s, l, term.Out, options)
+		// normal errors are already logged but not the panic recoveries
+		// Note this is the only case that can get contNeeded (EOL instead of EOF mode)
+		contNeeded, panicked, errs, formatted := EvalOne(s, l, term.Out, options)
 		if contNeeded {
 			prev = l + "\n"
 			term.SetPrompt(CONTINUATION)
 		} else {
-			if prev != "" {
+			if prev != "" && len(formatted) > 0 {
 				// In addition to raw lines, we also add the single line version to history.
 				log.LogVf("Adding to history: %q", formatted)
 				term.AddToHistory(formatted)
 			}
 			prev = ""
 			term.SetPrompt(PROMPT)
 		}
+		if panicked {
+			log.Errf("%s", errs[0]) // we know there is exactly 1 error in this case, see EvalOne defer.
+		}
```

Comment:
I don't get the need for this log as the defer recover already log the error

Reply:
you're right, I think it was something to do with showing up on same line as eval but... removed

before
```
$ 0/0
== Parse ==> 0/0
== Eval  ==> 09:07:43.886 [CRI] Caught panic: runtime error: integer divide by zero
09:07:43.886 [ERR] panic: runtime error: integer divide by zero
```

now
```
$ func (){self()}()
== Parse ==> func(){self()}()
== Eval  ==> 09:10:16.289 [CRI] Caught panic: max depth 250000 reached
```

(that's with -parse, without it looks ok)